### PR TITLE
Improve test coverage related to 'bracket-color'

### LINKED ISSUE #26: 테스트 코드 보강하기
라이브러리가 잘 동작하는지 확인하기 위해 테스트 코드를 사용합니다. 테스트 코드는 견고할수록 좋습니다. 현재 작성되어 있는 테스트 코드들을 확인하고 미처 확인하지 못한 동작이 있다면 추가해 봅시다.

### PR DESCRIPTION
**Pull request by neokkk**
## What
1. Added shared test utilities to reduce duplicated assertion logic.

2. Refactored and extended existing tests so that each module validates only its own responsibilities.
   - Removed overlapping color conversion tests between rgb.rs and hsv.rs
   - Added validation for RGBA alpha handling
   - Improved lerp and grayscale-related test coverage

3. Added missing edge case and error handling tests.
   - Added invalid length / invalid character / missing hash tests for from_hex
   - Added iterator termination and panic case validation

## Why
- To reduce duplicated test code and improve maintainability.
- To make test responsibilities clearer and improve readability of test intent.
- To improve robustness by covering previously untested error handling and edge cases.

### Required

- [x] `cargo check --all` passes
- [x] `cargo fmt --all -- --check` passes
- [x] `cargo clippy --workspace --all-targets -- -D warnings -A clippy::multiple-crate-versions` passes
- [x] `cargo test --all` passes
- [x] I linked the related issue (for example: `Closes #26`)

### Functional Validation

- [ ] Behavior related to this change was verified locally (if applicable)
- [ ] Rendering/backend behavior was verified when runtime code changed (if applicable)
- [ ] Algorithm behavior (pathfinding/FOV/noise/random) was verified when affected (if applicable)
- [ ] I added or updated tests for changed behavior (if applicable)

### Configuration & Docs

- [ ] User-facing docs were updated (`README.md`, `ARCHITECTURE.md`, or relevant manual pages, if applicable)
- [ ] New dependencies/configuration are documented (if applicable)
- [ ] No sensitive values or credentials were introduced

### If Applicable

- [ ] Security impact considered (run `cargo audit` locally if needed)
- [ ] Breaking behavior changes are clearly described in this PR

## Summary by CodeRabbit

* **New Features**
  * Added XpColor conversion support to RGB (to_xp/from_xp methods)
  * Enabled WebAssembly compatibility in examples
  * Introduced new xpcolor module to bracket-color

* **Bug Fixes**
  * Fixed ExactSizeIterator::len() calculation for interpolation iterators to report remaining elements correctly

* **Tests**
  * Expanded test coverage with parametrized tests and shared assertion utilities

* **Chores**
  * Added byteorder and rstest dependencies
  * Reorganized imports for improved code structure

[Review Change Stack](https://app.coderabbit.ai/change-stack/utilForever/roguekit/pull/68)

### CODE REVIEW

File: bracket-color/src/hsv.rs [hunk 1]
```diff
@@ -160,43 +161,60 @@ mod tests {
...
+        let cases = [
+            (HSV::from_f32(0.0, 1.0, 1.0), 1.0, 0.0, 0.0),
+            (HSV::from_f32(120.0 / 360.0, 1.0, 1.0), 0.0, 1.0, 0.0),
+            (HSV::from_f32(240.0 / 360.0, 1.0, 1.0), 0.0, 0.0, 1.0),
+        ];
+
+        for (hsv, r, g, b) in cases {
+            assert_rgb_eq(hsv.to_rgb(), r, g, b);
+        }
     }
```
Comment (utilForever) [inline bracket-color/src/hsv.rs]: `rstest`라는 라이브러리를 사용하면 훨씬 간단하게 코드를 구현할 수 있습니다. 한 번 살펴 보세요.
https://github.com/la10736/rstest

File: bracket-color/src/hsv.rs [hunk 2]
```diff
@@ -160,43 +161,60 @@ mod tests {
...
+        let cases = [
+            (HSV::from_f32(0.0, 0.0, 0.0), 0.0, 0.0, 0.0),
+            (HSV::from_f32(0.0, 0.0, 1.0), 1.0, 1.0, 1.0),
+            (HSV::from_f32(0.0, 0.0, 0.5), 0.5, 0.5, 0.5),
+        ];
+
+        for (hsv, r, g, b) in cases {
+            assert_rgb_eq(hsv.to_rgb(), r, g, b);
+        }
+    }
```
Comment (utilForever) [inline bracket-color/src/hsv.rs]: 여기도 `rstest`를 사용하면 간단하게 코드를 구현할 수 있습니다.

File: bracket-color/src/lerpit.rs [hunk 3]
```diff
@@ -241,3 +241,78 @@ impl Iterator for AlphaLerp {
...
+
+    #[test]
+    fn rgb_lerp_len_returns_configured_steps() {
+        let lerp = RgbLerp::new(
+            RGB::from_f32(0.0, 0.0, 0.0),
+            RGB::from_f32(1.0, 1.0, 1.0),
+            2,
+        );
+
+        assert_eq!(lerp.len(), 2);
```
Comment (utilForever) [inline bracket-color/src/lerpit.rs]: 위의 코드를 보면 `lerp`의 길이가 3이어야 될 거 같은데, 여기는 왜 2인가요?

File: bracket-color/src/lerpit.rs [hunk 4]
```diff
@@ -241,3 +241,78 @@ impl Iterator for AlphaLerp {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
```
Comment (utilForever) [inline bracket-color/src/lerpit.rs:246]: 테스트 코드에 있는 `Lerp`의 `steps`가 전부 2입니다. `steps`를 서로 다른 값으로 바꾸는 게 좋을 거 같습니다.

File: bracket-color/src/rgb.rs [hunk 5]
```diff
@@ -462,49 +463,98 @@ mod tests {
...
+        let cases = [
+            (RGB::from_f32(1.0, 0.0, 0.0), 0.0, 1.0, 1.0),
+            (RGB::from_f32(0.0, 1.0, 0.0), 120.0 / 360.0, 1.0, 1.0),
+            (RGB::from_f32(0.0, 0.0, 1.0), 240.0 / 360.0, 1.0, 1.0),
+        ];
+
+        for (rgb, h, s, v) in cases {
+            assert_hsv_eq(rgb.to_hsv(), h, s, v);
+        }
+    }
```
Comment (utilForever) [inline bracket-color/src/rgb.rs]: 여기도 `rstest`를 사용하면 간단하게 코드를 구현할 수 있습니다.

File: bracket-color/src/rgb.rs [hunk 6]
```diff
@@ -462,49 +463,98 @@ mod tests {
...
+        let cases = [
+            (RGB::from_f32(0.0, 0.0, 0.0), 0.0, 0.0, 0.0),
+            (RGB::from_f32(1.0, 1.0, 1.0), 0.0, 0.0, 1.0),
+            (RGB::from_f32(0.5, 0.5, 0.5), 0.0, 0.0, 0.5),
+        ];
+
+        for (rgb, h, s, v) in cases {
+            assert_hsv_eq(rgb.to_hsv(), h, s, v);
+        }
+    }
```
Comment (utilForever) [inline bracket-color/src/rgb.rs]: 여기도 `rstest`를 사용하면 간단하게 코드를 구현할 수 있습니다.

File: bracket-color/src/rgb.rs [hunk 7]
```diff
@@ -462,49 +463,98 @@ mod tests {
...
+            ("#00FF00", 0.0, 1.0, 0.0),
+            ("#0000FF", 0.0, 0.0, 1.0),
+            ("#808000", 128.0 / 255.0, 128.0 / 255.0, 0.0),
+        ];
+
+        for (hex, r, g, b) in cases {
+            let rgb = RGB::from_hex(hex).expect("valid hex color");
+            assert_rgb_eq(rgb, r, g, b);
+        }
+    }
```
Comment (utilForever) [inline bracket-color/src/rgb.rs]: 여기도 `rstest`를 사용하면 간단하게 코드를 구현할 수 있습니다.

File: bracket-color/src/rgb.rs [hunk 8]
```diff
@@ -462,49 +463,98 @@ mod tests {
...
+
+    #[test]
+    fn parse_hex_rejects_invalid_length() {
+        let cases = ["", "#FFF", "#FFFF", "#FFFFFF00"];
+
+        for hex in cases {
+            let err = RGB::from_hex(hex).unwrap_err();
+            assert_eq!(err, HtmlColorConversionError::InvalidStringLength);
+        }
+    }
```
Comment (utilForever) [inline bracket-color/src/rgb.rs]: 여기도 `rstest`를 사용하면 간단하게 코드를 구현할 수 있습니다.

File: bracket-color/src/rgba.rs [hunk 9]
```diff
@@ -487,54 +488,95 @@ mod tests {
...
+                0.0,
+                128.0 / 255.0,
+            ),
+        ];
+
+        for (hex, r, g, b, a) in cases {
+            let rgba = RGBA::from_hex(hex).expect("valid RGBA hex color");
+            assert_rgba_eq(rgba, r, g, b, a);
+        }
+    }
```
Comment (utilForever) [inline bracket-color/src/rgba.rs]: 여기도 `rstest`를 사용하면 간단하게 코드를 구현할 수 있습니다.

File: bracket-color/src/rgba.rs [hunk 10]
```diff
@@ -487,54 +488,95 @@ mod tests {
...
-        let rgb = RGBA::from_hex("#00FF00FF").expect("Invalid hex string");
-        assert!(rgb.r < f32::EPSILON);
-        assert!(f32::abs(rgb.g - 1.0) < f32::EPSILON);
-        assert!(rgb.b < f32::EPSILON);
-        assert!((rgb.a - 1.0).abs() < f32::EPSILON);
+    fn from_u8_converts_components() {
+        let rgba = RGBA::from_u8(255, 128, 0, 64);
+
+        assert_rgba_eq(rgba, 1.0, 128.0 / 255.0, 0.0, 64.0 / 255.0);
     }
```
Comment (utilForever) [inline bracket-color/src/rgba.rs:506]:
```suggestion
    fn from_u8_converts_components() {
        let rgba = RGBA::from_u8(255, 128, 0, 64);
        assert_rgba_eq(rgba, 1.0, 128.0 / 255.0, 0.0, 64.0 / 255.0);
}
```

File: bracket-color/src/rgba.rs [hunk 11]
```diff
@@ -487,54 +488,95 @@ mod tests {
...
+
+    #[test]
+    fn parse_hex_rejects_invalid_length() {
+        let cases = ["", "#FFF", "#FFFFFF", "#FFFFFFFF00"];
+
+        for hex in cases {
+            let err = RGBA::from_hex(hex).unwrap_err();
+            assert_eq!(err, HtmlColorConversionError::InvalidStringLength);
+        }
+    }
```
Comment (utilForever) [inline bracket-color/src/rgba.rs]: 여기도 `rstest`를 사용하면 간단하게 코드를 구현할 수 있습니다.

File: bracket-color/src/xpcolor.rs [hunk 12]
```diff
@@ -72,3 +72,69 @@ impl From<XpColor> for RGB {
...
+    }
+
+    #[test]
+    fn read_xp_color_reads_three_bytes() {
+        let mut input = &[1_u8, 2, 3][..];
+
+        let color = XpColor::read(&mut input).expect("valid RGB bytes");
+
+        assert_eq!(color, XpColor::new(1, 2, 3));
+    }
```
Comment (utilForever) [inline bracket-color/src/xpcolor.rs:103]:
```suggestion
    #[test]
    fn read_xp_color_reads_three_bytes() {
        let mut input = &[1_u8, 2, 3][..];
        let color = XpColor::read(&mut input).expect("valid RGB bytes");

        assert_eq!(color, XpColor::new(1, 2, 3));
    }
```

File: bracket-color/src/xpcolor.rs [hunk 13]
```diff
@@ -72,3 +72,69 @@ impl From<XpColor> for RGB {
...
+
+    #[test]
+    fn read_xp_color_returns_error_on_short_input() {
+        let mut input = &[1_u8, 2][..];
+
+        let result = XpColor::read(&mut input);
+
+        assert!(result.is_err());
+    }
```
Comment (utilForever) [inline bracket-color/src/xpcolor.rs:111]:
```suggestion
    #[test]
    fn read_xp_color_returns_error_on_short_input() {
        let mut input = &[1_u8, 2][..];
        let result = XpColor::read(&mut input);

        assert!(result.is_err());
    }
```

File: bracket-color/src/xpcolor.rs [hunk 14]
```diff
@@ -72,3 +72,69 @@ impl From<XpColor> for RGB {
...
+
+    #[test]
+    fn write_xp_color_writes_three_bytes() {
+        let color = XpColor::new(1, 2, 3);
+        let mut output = Vec::new();
+
+        color.write(&mut output).expect("write should succeed");
+
+        assert_eq!(output, vec![1, 2, 3]);
+    }
```
Comment (utilForever) [inline bracket-color/src/xpcolor.rs:121]:
```suggestion
    #[test]
    fn write_xp_color_writes_three_bytes() {
        let color = XpColor::new(1, 2, 3);
        let mut output = Vec::new();

        color.write(&mut output).expect("write should succeed");
        assert_eq!(output, vec![1, 2, 3]);
    }
```

**Review by utilForever — CHANGES_REQUESTED** [review]
Can you check my comments? Thanks.

**Comment by utilForever** [comment]
그리고 해당하는 이슈를 찾아서 `Closes #number.`으로 남겨주세요. 잘 모르겠으면 다른 PR 참고하세요.

File: bracket-color/src/lib.rs [hunk 15]
```diff
@@ -53,4 +58,5 @@ pub mod prelude {
     pub use crate::palette::*;
     pub use crate::rgb::*;
     pub use crate::rgba::*;
+    pub use crate::xpcolor::*;
```
Comment (neokkk) [inline bracket-color/src/lib.rs:61]: @utilForever `cargo check`랑 `cargo clippy` 수행 시 bracket-color가 아닌 terminal이나 rex에서 중복 모듈 관련 warning, error가 뜨는데 어떻게 해결하면 좋을까요?

Comment (utilForever) [reply bracket-color/src/lib.rs:61]: 이건 수정하기가 쉽지 않은 이슈라 제가 해결할게요.

File: bracket-color/src/lerpit.rs [hunk 16]
```diff
@@ -241,3 +241,106 @@ impl Iterator for AlphaLerp {
...
+        assert!(lerp.next().is_none());
+    }
+
+    #[rstest]
+    #[case(1, vec![0.0, 1.0])]
+    #[case(2, vec![0.0, 0.5, 1.0])]
+    #[case(4, vec![0.0, 0.25, 0.5, 0.75, 1.0])]
+    fn rgba_lerp_iterates_from_start_to_end(
+        #[case] steps: usize,
+
```
Comment (utilForever) [inline bracket-color/src/lerpit.rs]: 불필요한 빈 줄 삭제해 주세요.

File: bracket-color/src/lib.rs [hunk 17]
```diff
@@ -42,6 +42,11 @@ pub mod palette;
 pub mod rgb;
 /// Import RGBA color support
 pub mod rgba;
+/// Import color wrapper for transportation support
+pub mod xpcolor;
+
+#[cfg(test)]
+pub mod test_utils;
```
Comment (utilForever) [inline bracket-color/src/lib.rs]: 테스트용으로만 쓰는 건데 `pub`일 필요가 없습니다.

```suggestion
pub(crate) mod test_utils;
```

File: bracket-color/src/test_utils.rs [hunk 18]
```diff
@@ -0,0 +1,28 @@
+use crate::prelude::{HSV, RGB, RGBA};
+
+#[allow(clippy::missing_panics_doc)]
+pub fn assert_approx_eq(left: f32, right: f32) {
```
Comment (utilForever) [inline bracket-color/src/test_utils.rs]:
```suggestion
pub(crate) fn assert_approx_eq(left: f32, right: f32) {
```

File: bracket-color/src/test_utils.rs [hunk 19]
```diff
@@ -0,0 +1,28 @@
...
+    assert!((left - right).abs() < 1e-6);
+}
+
+#[allow(clippy::missing_panics_doc)]
+pub fn assert_rgb_eq(rgb: RGB, r: f32, g: f32, b: f32) {
```
Comment (utilForever) [inline bracket-color/src/test_utils.rs]:
```suggestion
pub(crate) fn assert_rgb_eq(rgb: RGB, r: f32, g: f32, b: f32) {
```

File: bracket-color/src/test_utils.rs [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+    assert_approx_eq(rgb.r, r);
+    assert_approx_eq(rgb.g, g);
+    assert_approx_eq(rgb.b, b);
+}
+
+#[allow(clippy::missing_panics_doc)]
+pub fn assert_rgba_eq(rgba: RGBA, r: f32, g: f32, b: f32, a: f32) {
```
Comment (utilForever) [inline bracket-color/src/test_utils.rs]:
```suggestion
pub(crate) fn assert_rgba_eq(rgba: RGBA, r: f32, g: f32, b: f32, a: f32) {
```

File: bracket-color/src/test_utils.rs [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+    assert_approx_eq(rgba.r, r);
+    assert_approx_eq(rgba.g, g);
+    assert_approx_eq(rgba.b, b);
+    assert_approx_eq(rgba.a, a);
+}
+
+#[allow(clippy::missing_panics_doc)]
+pub fn assert_hsv_eq(hsv: HSV, h: f32, s: f32, v: f32) {
```
Comment (utilForever) [inline bracket-color/src/test_utils.rs]:
```suggestion
pub(crate) fn assert_hsv_eq(hsv: HSV, h: f32, s: f32, v: f32) {
```

File: bracket-color/src/xpcolor.rs [hunk 22]
```diff
@@ -72,3 +72,69 @@ impl From<XpColor> for RGB {
...
+        assert!(XpColor::TRANSPARENT.is_transparent());
+        assert!(!XpColor::BLACK.is_transparent());
+        assert!(!XpColor::new(255, 0, 254).is_transparent());
+    }
+
+    #[test]
+    fn read_xp_color_reads_three_bytes() {
+        let mut input = &[1_u8, 2, 3][..];
+
+        let color = XpColor::read(&mut input).expect("valid RGB bytes");
```
Comment (utilForever) [inline bracket-color/src/xpcolor.rs:100]:
```suggestion
        let mut input = &[1_u8, 2, 3][..];
        let color = XpColor::read(&mut input).expect("valid RGB bytes");
```

File: bracket-color/src/xpcolor.rs [hunk 23]
```diff
@@ -72,3 +72,69 @@ impl From<XpColor> for RGB {
...
+
+        assert_eq!(color, XpColor::new(1, 2, 3));
+    }
+
+    #[test]
+    fn read_xp_color_returns_error_on_short_input() {
+        let mut input = &[1_u8, 2][..];
+
+        let result = XpColor::read(&mut input);
```
Comment (utilForever) [inline bracket-color/src/xpcolor.rs:108]:
```suggestion
        let mut input = &[1_u8, 2][..];
        let result = XpColor::read(&mut input);
```

**Review by utilForever — CHANGES_REQUESTED** [review]
다음 사항들도 수정 부탁드립니다.

File: bracket-color/src/lib.rs (hunk 15 above)
Comment (utilForever) [reply bracket-color/src/lib.rs:61]: 수정했습니다.

File: bracket-terminal/src/lib.rs [hunk 24]
```diff
@@ -31,7 +31,7 @@ pub mod prelude {
     pub use crate::initializer::*;
     pub use crate::input::{BEvent, Input, INPUT};
     pub use crate::rex;
-    pub use crate::rex::*;
+    pub use crate::rex::{xp_to_console, xp_to_draw_batch, XpCell, XpFile, XpLayer};
```
[the remark this answers is not in the document]
Comment (utilForever) [reply bracket-terminal/src/lib.rs]: I think it is resolved.

File: bracket-terminal/src/rex.rs [hunk 25]
```diff
@@ -1,7 +1,7 @@
 use crate::prelude::{Console, DrawBatch, FontCharType};
 use bracket_color::prelude::{ColorPair, RGBA};
 use bracket_geometry::prelude::Point;
-pub use bracket_rex::prelude::{XpCell, XpColor, XpFile, XpLayer};
+use bracket_rex::prelude::XpFile;
```
[the remark this answers is not in the document]
Comment (utilForever) [reply bracket-terminal/src/rex.rs:4]: Yes, this is intentional.

**Review by utilForever — APPROVED** [review]
LGTM now. Thanks for your work! 🎉
